src/main/java/org/folio/search/utils/CallNumberUtils.java

```diff
@@ -144,6 +147,25 @@ public static Long getCallNumberAsLong(String callNumber, int firstPosition) {
     return callNumberToLong(callNumber, startVal, CN_MAX_CHARS - 1);
   }
 
+  public static List<CallNumberBrowseItem> excludeIrrelevantResultItems(String refinedCondition,
+                                                                        List<CallNumberBrowseItem> records) {
+    if (StringUtils.isBlank(refinedCondition)) {
+      return records;
+    }
+
+    records = records
+      .stream()
+      .filter(distinctByKey(r -> r.getInstance().getId()))
+      .toList();
+    records
+      .forEach(r -> r.getInstance().setItems(r.getInstance().getItems()
+        .stream()
+        .filter(i -> CallNumberType.fromId(i.getEffectiveCallNumberComponents().getTypeId())
+          .equals(CallNumberType.fromName(refinedCondition)))
+        .toList()));
```

Comment:
Looks like we can do it in one stream sequence.
And why do wee need to filter records by instanceId at all?